Automattic/woocommerce-payments
# Ensure WCPay JS loaded on Checkout page for zero order totals

When an item with price 0 is added to the cart, it is expected that the payment form does not need to be displayed on the Checkout page. But, when the cart is updated via AJAX on the checkout page to be non-zero, the payment form should appear, because it is no longer a 0 priced order.

Fixes #4764
Fixes #5277

#### Changes proposed in this Pull Request
- Even when cart is zero, make sure the JS is loaded.
- Do this only on checkout page, and not on blocks page because the Blocks Checkout already loads all the required JS. 

#### Testing instructions
1) Test with steps in #4764 
2) Test with steps in #5277 
3) Test with Checkout Blocks and UPE (all variations of checkout) and on other pages where the checkout form is used e.g. Pay for Order, My Accounts etc to ensure no regression 
4) Can test with any plugins that update order via AJAX on checkout page.

-------------------


**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-5.6.0#ensure-wcpay-js-loaded-on-checkout-page-for-zero-order-totals-pr-5423
- [x] Not applicable - Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
